ballerina-tests/tests/data_binding_payload_tests.bal

```diff
@@ -0,0 +1,666 @@
+    lock {
+        receivedFloatPayloadValuePublish = message;
+    }
+}
+
+isolated function getReceivedFloatPayloadValuePublish() returns float {
+    lock {
+        return receivedFloatPayloadValuePublish;
+    }
+}
+
+isolated function setReceivedDecimalPayloadValuePublish(decimal message) {
+    lock {
+        receivedDecimalPayloadValuePublish = message;
+    }
+}
+
+isolated function getReceivedDecimalPayloadValuePublish() returns decimal {
+    lock {
+        return receivedDecimalPayloadValuePublish;
+    }
+}
+
+isolated function setReceivedBooleanPayloadValuePublish(boolean message) {
+    lock {
+        receivedBooleanPayloadValuePublish = message;
+    }
+}
+
+isolated function getReceivedBooleanPayloadValuePublish() returns boolean {
+    lock {
+        return receivedBooleanPayloadValuePublish;
+    }
+}
+
+isolated function setReceivedPersonPayloadValuePublish(Person? message) {
+    lock {
+        receivedPersonPayloadValuePublish = message.clone();
+    }
+}
+
+isolated function getReceivedPersonPayloadValuePublish() returns Person? {
+    lock {
+        return receivedPersonPayloadValuePublish.clone();
+    }
+}
+
+isolated function setReceivedMapPayloadValuePublish(map<Person> message) {
+    lock {
+        receivedMapPayloadValuePublish = message.clone();
+    }
+}
+
+isolated function getReceivedMapPayloadValuePublish() returns map<Person>? {
+    lock {
+        return receivedMapPayloadValuePublish.clone();
+    }
+}
+
+isolated function setReceivedTablePayloadValuePublish(table<Person> message) {
+    lock {
+        receivedTablePayloadValuePublish = message.clone();
+    }
+}
+
+isolated function getReceivedTablePayloadValuePublish() returns table<Person>? {
+    lock {
+        return receivedTablePayloadValuePublish.clone();
+    }
+}
+
+isolated function setReceivedUnionPayloadValuePublish(int|string message) {
+    lock {
+        receivedUnionPayloadValuePublish = message.clone();
+    }
+}
+
+isolated function getReceivedUnionPayloadValuePublish() returns int|string? {
+    lock {
+        return receivedUnionPayloadValuePublish.clone();
+    }
+}
+
+isolated function setReceivedRandomPayloadValuePublish(RandomPayload message) {
+    lock {
+        receivedRandomPayloadValuePublish = message.clone();
+    }
+}
+
+isolated function getReceivedRandomPayloadValuePublish() returns RandomPayload? {
+    lock {
+        return receivedRandomPayloadValuePublish.clone();
+    }
+}
+
+public type RandomPayload record {|
+    string content;
+    string subject;
+    string replyTo?;
+|};
+
+@test:Config {
+    groups: ["nats-basic"]
+}
+public function testDataBindingStringPayloadPublish() returns error? {
+    string messageToSend = "Testing Consumer Service With String Data Binding.";
+    nats:Client newClient = check new(DATA_BINDING_URL);
+    check newClient->publishMessage({ content: messageToSend, subject: "dataStringPayload" });
+    int timeoutInSeconds = 120;
```

Comment:
Shall we define this as a module-level variable, so that we don't need to define it in each test function? We should also consider removing this waiting logic altogether.